polynote-kernel/src/main/scala/polynote/config/PolynoteConfig.scala

```diff
@@ -25,13 +25,23 @@ object Storage {
   implicit val decoder: Decoder[Storage] = deriveDecoder
 }
 
+final case class Behavior(
+  dependencyIsolation: Boolean = true
+)
+
+object Behavior {
+  implicit val encoder: ObjectEncoder[Behavior] = deriveEncoder
+  implicit val decoder: Decoder[Behavior] = deriveDecoder
+}
+
 final case class PolynoteConfig(
   listen: Listen = Listen(),
   storage: Storage = Storage(),
   repositories: List[RepositoryConfig] = Nil,
   exclusions: List[String] = Nil,
   dependencies: Map[String, List[String]] = Map.empty,
-  spark: Map[String, String] = Map.empty
+  spark: Map[String, String] = Map.empty,
+  behavior: Behavior = Behavior()
```

Comment:
Maybe add an entry to `config-template.yml` describing the usage of this new key?

Reply:
Also, would you want to fold other property-based config into `Behavior` (like `polynote.debug`)?

Reply:
yes, probably at some point that would make sense. I don't to publicize this option yet as we might change it around (`behavior` is kind of a bad name)